android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/StringExtensions.kt

```diff
@@ -42,15 +43,19 @@ fun String.interpolate(
   prefix: String = "@{",
   suffix: String = "}"
 ): String =
-  StringSubstitutor.replace(
-    this.replace(Pattern.quote(prefix).plus(".*?").plus(Pattern.quote(suffix)).toRegex()) {
-      it.value.replace("\\s+".toRegex(), "")
-    },
-    lookupMap,
-    prefix,
-    suffix
-  )
-
+  try {
+    StringSubstitutor.replace(
+      this.replace(Pattern.quote(prefix).plus(".*?").plus(Pattern.quote(suffix)).toRegex()) {
+        it.value.replace("\\s+".toRegex(), "")
+      },
+      lookupMap,
+      prefix,
+      suffix
+    )
+  } catch (e: Exception) {
```

Comment:
Is there anything more specific to catch? Can we test an input that leads to this exception?

Reply:
From the docs the function throws `IllegalArgumentException` [StringSubtitutor#replace](https://commons.apache.org/proper/commons-text/apidocs/org/apache/commons/text/StringSubstitutor.html#replace-java.lang.Object-java.util.Map-java.lang.String-java.lang.String-)

Reply:
We could narrow it down I think to respective Exceptions but basically any exception thrown here will cause the app to crash

Reply:
I'm not that familiar with the structure of Android apps, but Isn't that (if an exception is thrown the app will crash) true of a lot of the code in a lot of places?

Is there some higher up main loop where we can can catch all Exceptions? So the app usually returns an error message to the user when there's an unrecoverable error, instead of crashing.

And then here we catch exceptions that we expect to be thrown here.

Reply:
Yeah you're correct in that it will crash unless caught higher up. That is unless we catch the exception in the function it will have to be done at each point of invocation. 

For a global all catch solution doing some research on that

Reply:
moved to this issue, https://github.com/opensrp/fhircore/issues/2156